src/sync/sync_manager.cpp

```diff
@@ -130,6 +154,34 @@ void SyncManager::configure_file_system(const std::string& base_file_path,
     }
 }
 
+bool SyncManager::immediately_run_file_actions(const std::string& realm_path)
+{
+    SyncFileActionMetadataResults file_actions = m_metadata_manager->all_pending_actions();
+    for (size_t i = 0; i < file_actions.size(); i++) {
+        SyncFileActionMetadata file_action = file_actions.get(i);
+        if (file_action.original_name() != realm_path) {
```

Comment:
Rather than retrieving all actions and looping to find one with a matching path, why not just look the action up directly by path?